docs/guides/developer/docs/reviewing-and-merging.md

```diff
@@ -44,21 +43,19 @@ all pull requests are to ensure that:
     * [Licenses seem to be fine](license.md)
     * Licenses are added to the `NOTICES` file
 * Feature documentation is in place
-* An upgrade path exists (usually relevant for database changes)
-
-
-The exact review process heavily depends on the type, size and purpose of the pull request.
+* Adopters can easily upgrade to new versions
+    * An upgrade path exists
+    * The upgrade path is documented
 
 Here are some things a reviewer should usually do:
 
 * Check if code looks sensible (e.g. no nonsensical files, no obvious formatting errors, no large binary files, …)
 * Locally run Opencast and verify that the issue has been fixed (as described in the pull request and/or the Jira
   ticket) or the feature does what it is supposed
+* Check whether the database schema is modified. In case it is, a database migration scripts needs to be provided
+  and the upgrade guide needs to contain the information that a database migration is needed including a reference
+  to the Jira ticket(s) the cause the migration
+* Check whether configuration files have been changed in a way that potentially requires Adopters to adjust their
+  configuration. Such configuration changes need to be documented in the upgrade guide including a reference to
```

Comment:
Right now, the documentation is mostly in the [release notes](https://docs.opencast.org/r/7.x/admin/releasenotes/#configuration-changes) which I think makes sense to keep the upgrade guide concise. Also, -1 to linking the Jira tickets. It might make sense but doesn't always, so just leave it out.

Reply:
... but this information is required when upgrading, so why should it be in the release notes?

-> Discussion should happen on list

Reply:
change logs and upgrade guides are sub-items of the release notes so
they are in there either way ;) …which makes sense since they are notes
about the release after all.

If you feel strongly about it, we can move it but historically, the
upgrade guide always contained a point which said more or less “compare
the old and new configuration” which is exactly what you need to do and
what you need to know and what I (and from comments on list I think
others as well) are doing. It's much faster than reading though
specific changes with many of them usually being irrelevant.

That's why my tendency would be to keep the detailed list of changes in
the release notes. People can read them if they want to and reference
them if they need to, but they don't clutter the upgrade guide which in
turn means that the important points are less likely to be overlooked.

But I do not have a strong feeling about this and we can put the notes
in the upgrade section if people prefer that. In the end, it's the
release manager's decision anyway how to exactly structure the release
notes and if to copy things to a specific place.

Reply:
IMO, in a prod context, if you're not using a change management system for anything at scale then you're doing it wrong.  I'm sure we have adopters who *do* run everything by hand, but at some point they're going to feel the pain of that regardless of how well we document things.  Comparing their config to the new one is the only viable way for them to know that what they have is right for them.  As Lars said, our docs are going to list a ton of changes which are not relevant - how will they know what's relevant to them, since they're 'just' using the system rather than developing.  This becomes easier with staging environments, but even if they don't have that they'll still need to mark up the new config for their CM system.

Reply:
> IMO, in a prod context, if you're not using a change management system for anything at scale then you're doing it wrong. I'm sure we have adopters who do run everything by hand, but at some point they're going to feel the pain of that regardless of how well we document things

In our case we have an own copy of the full config which is in a separate GIT repository and managed/deployed using Ansible.
I am currently doing diffs between the configs to adjust them but consider this a quite error prone process: There are quite a large number of differences over the overall config between ours and the Opencast default configuration and it is easy to miss those differences.

> As Lars said, our docs are going to list a ton of changes which are not relevant

I don't think we would end-up a large list of irrelevant changes as there are not that many config changes then upgrading from one version to the next one. And the irrelevant changes also need to be looked at when going the "compare" way.

>  how will they know what's relevant to them, since they're 'just' using the system rather than developing.

The target audience of the upgrade guide is people that are performing the upgrades. Most likely they are also responsible for their Opencast configuration and hopefully know their configuration.

> This becomes easier with staging environments, but even if they don't have that they'll still need to mark up the new config for their CM system.

Imagine 6.x to 7.x had a complete list of configuration changes which are about 10 changes. Applying those to the CM is quite a lot less effort than doing a full diff on the full config which has several hundred difference which each need to be carefully looked at everytime.

Reply:
> I don't think we would end-up a large list of irrelevant changes as there are not that many config changes then upgrading from one version to the next one.

I imagine that there are 10 changes and then suddenly wake up with the realization that it's far away from reality 😸

Between 6 and 7, we have changes in 60 configuration files. `custom.properties` alone has 10 changes in properties. Of course, I can read about all of those changes and then apply them manually but that is very tedious work and I wouldn't do that.

Instead, I would look at the differences which lets me quickly apply those changes and be done with it. Of course, I wouldn't need to apply the change if there is a sensible default, but that's not always guaranteed (although I like that to be the case) and it's hard to keep track of my actual changes if I do not sync with the changes.

Or, depending on how many changes I made in the first place, I might even go with the default configuration and just apply those few changes.

But that may very well just be a difference in upgrade/administration style resulting in you wanting everything and me wanting an abbreviated version in the docs with data changes only.